In [1]:
!pip install kaggle


# Big Data Analysis Using PySpark  
### Internship Task 1 – CODTECH  

**Name:** Your Name  
**Role:** Data Analytics Intern  
**Tool Used:** PySpark  


## Objective

The objective of this task is to perform scalable data analysis using PySpark on a structured e-commerce dataset.  
This task demonstrates the ability to process large datasets and derive meaningful business insights using distributed computing.



## Dataset Description

The dataset contains 30,600 records of Indian e-commerce transactions.  

Main features include:
- Order ID and Order Date
- State and Zone
- Product Category
- Brand Type (Mass / Premium)
- Customer Gender and Age
- Base Price, Discount, Final Price
- Units Sold and Revenue
- Sales Event (Festival / Normal)
- Competition Intensity
- Inventory Pressure



In [2]:
from google.colab import files
files.upload()


Saving Indian_Ecommerce(in).csv to Indian_Ecommerce(in).csv


{'Indian_Ecommerce(in).csv': b'order_id,order_date,state,zone,category,brand_type,customer_gender,customer_age,base_price,discount_percent,final_price,units_sold,revenue,sales_event,competition_intensity,inventory_pressure\r\nORD0000001,1/31/2023,West Bengal,East,Fashion,Mass,Male,28,1810.89,65,633.81,29,18380.49,Normal,Medium,High\r\nORD0000002,1/31/2023,Gujarat,West,Sports & Fitness,Mass,Male,19,5678.15,50.83,2791.95,67,187060.65,Normal,Medium,High\r\nORD0000003,1/31/2023,Delhi NCR,North,Grocery Essentials,Mass,Male,25,169.98,35.26,110.05,57,6272.85,Normal,Medium,Low\r\nORD0000004,1/31/2023,Madhya Pradesh,Central,Footwear,Mass,Female,30,3244.91,61.94,1235.01,55,67925.55,Normal,High,High\r\nORD0000005,1/31/2023,Haryana,North,Fashion,Premium,Female,33,697.44,48.27,360.79,29,10462.91,Festival,Medium,High\r\nORD0000006,1/31/2023,Punjab,North,Grocery Essentials,Mass,Female,19,100.55,65,35.19,16,563.04,Festival,High,High\r\nORD0000007,1/31/2023,Delhi NCR,North,Home & Living,Mass,Female,47,

In [18]:
# Import SparkSession
from pyspark.sql import SparkSession

# Create Spark session
spark = SparkSession.builder \
    .appName("Indian Ecommerce Big Data Analysis") \
    .getOrCreate()

print("Spark Session Created Successfully")


Spark Session Created Successfully


## Data Loading

The dataset is loaded into a PySpark DataFrame for scalable processing.



In [19]:
# Load CSV file into Spark DataFrame
df = spark.read.csv("Indian_Ecommerce(in).csv", header=True, inferSchema=True)

print("Total Records:", df.count())


Total Records: 30600


## Schema Overview

Inspecting the structure and data types of the dataset.



In [20]:
df.printSchema()
df.show(5)


root
 |-- order_id: string (nullable = true)
 |-- order_date: string (nullable = true)
 |-- state: string (nullable = true)
 |-- zone: string (nullable = true)
 |-- category: string (nullable = true)
 |-- brand_type: string (nullable = true)
 |-- customer_gender: string (nullable = true)
 |-- customer_age: integer (nullable = true)
 |-- base_price: double (nullable = true)
 |-- discount_percent: double (nullable = true)
 |-- final_price: double (nullable = true)
 |-- units_sold: integer (nullable = true)
 |-- revenue: double (nullable = true)
 |-- sales_event: string (nullable = true)
 |-- competition_intensity: string (nullable = true)
 |-- inventory_pressure: string (nullable = true)

+----------+----------+--------------+-------+------------------+----------+---------------+------------+----------+----------------+-----------+----------+---------+-----------+---------------------+------------------+
|  order_id|order_date|         state|   zone|          category|brand_type|customer

In [4]:
import os
os.listdir()


['.config', 'Indian_Ecommerce(in).csv', 'sample_data']

In [6]:
!pip install pyspark


In [7]:
from pyspark.sql import SparkSession

spark = SparkSession.builder \
    .appName("Indian Ecommerce Big Data Analysis") \
    .getOrCreate()

print("Spark Session Created Successfully")


Spark Session Created Successfully


In [9]:
import os
os.listdir()


['.config', 'Indian_Ecommerce(in).csv', 'sample_data']

In [10]:
df = spark.read.csv("Indian_Ecommerce(in).csv", header=True, inferSchema=True)

df.printSchema()
print("Total Records:", df.count())


root
 |-- order_id: string (nullable = true)
 |-- order_date: string (nullable = true)
 |-- state: string (nullable = true)
 |-- zone: string (nullable = true)
 |-- category: string (nullable = true)
 |-- brand_type: string (nullable = true)
 |-- customer_gender: string (nullable = true)
 |-- customer_age: integer (nullable = true)
 |-- base_price: double (nullable = true)
 |-- discount_percent: double (nullable = true)
 |-- final_price: double (nullable = true)
 |-- units_sold: integer (nullable = true)
 |-- revenue: double (nullable = true)
 |-- sales_event: string (nullable = true)
 |-- competition_intensity: string (nullable = true)
 |-- inventory_pressure: string (nullable = true)

Total Records: 30600


In [11]:
df.printSchema()


root
 |-- order_id: string (nullable = true)
 |-- order_date: string (nullable = true)
 |-- state: string (nullable = true)
 |-- zone: string (nullable = true)
 |-- category: string (nullable = true)
 |-- brand_type: string (nullable = true)
 |-- customer_gender: string (nullable = true)
 |-- customer_age: integer (nullable = true)
 |-- base_price: double (nullable = true)
 |-- discount_percent: double (nullable = true)
 |-- final_price: double (nullable = true)
 |-- units_sold: integer (nullable = true)
 |-- revenue: double (nullable = true)
 |-- sales_event: string (nullable = true)
 |-- competition_intensity: string (nullable = true)
 |-- inventory_pressure: string (nullable = true)



In [12]:
df.show(5)


+----------+----------+--------------+-------+------------------+----------+---------------+------------+----------+----------------+-----------+----------+---------+-----------+---------------------+------------------+
|  order_id|order_date|         state|   zone|          category|brand_type|customer_gender|customer_age|base_price|discount_percent|final_price|units_sold|  revenue|sales_event|competition_intensity|inventory_pressure|
+----------+----------+--------------+-------+------------------+----------+---------------+------------+----------+----------------+-----------+----------+---------+-----------+---------------------+------------------+
|ORD0000001| 1/31/2023|   West Bengal|   East|           Fashion|      Mass|           Male|          28|   1810.89|            65.0|     633.81|        29| 18380.49|     Normal|               Medium|              High|
|ORD0000002| 1/31/2023|       Gujarat|   West|  Sports & Fitness|      Mass|           Male|          19|   5678.15|    

In [13]:
from pyspark.sql.functions import sum

df.select(sum("revenue")).show()


+--------------------+
|        sum(revenue)|
+--------------------+
|2.1745103669900055E9|
+--------------------+



In [14]:
df.groupBy("sales_event") \
  .sum("revenue") \
  .show()


+-----------+--------------------+
|sales_event|        sum(revenue)|
+-----------+--------------------+
|   Festival| 7.606892495400027E8|
|     Normal|1.4138211174499946E9|
+-----------+--------------------+



In [15]:
df.groupBy("category") \
  .sum("revenue") \
  .orderBy("sum(revenue)", ascending=False) \
  .show()


+--------------------+--------------------+
|            category|        sum(revenue)|
+--------------------+--------------------+
|   Premium Lifestyle|      6.4809691069E8|
|         Electronics| 5.205174911400006E8|
|    Sports & Fitness| 2.833666460400004E8|
|       Home & Living|2.3667829913000074E8|
|            Footwear|1.9576828262999964E8|
|             Fashion|1.5787525019999993E8|
|Beauty & Personal...|1.0162320439999971E8|
|  Grocery Essentials|3.0584282760000054E7|
+--------------------+--------------------+



In [16]:
df.groupBy("state") \
  .sum("revenue") \
  .orderBy("sum(revenue)", ascending=False) \
  .show()


+--------------+--------------------+
|         state|        sum(revenue)|
+--------------+--------------------+
|     Delhi NCR|1.9569107363000003E8|
|   Maharashtra|1.8944358546000004E8|
|     Karnataka|1.8876721442000037E8|
|        Kerala| 1.683533524499998E8|
|       Gujarat|1.6821556560000008E8|
|     Telangana| 1.669117726399997E8|
|    Tamil Nadu|1.6540152774000007E8|
|        Punjab|1.5866575261999977E8|
|       Haryana|1.5723338196000013E8|
|Madhya Pradesh|1.3122563678999998E8|
| Uttar Pradesh|1.2492213223999994E8|
|   West Bengal|1.2246277585999976E8|
|        Odisha| 1.218676474700002E8|
|  Chhattisgarh|1.1534894810999987E8|
+--------------+--------------------+



In [17]:
df.groupBy("customer_gender")
  .sum("revenue") \
  .show()


+---------------+--------------------+
|customer_gender|        sum(revenue)|
+---------------+--------------------+
|         Female|1.0884523070900016E9|
|           Male| 1.086058059899995E9|
+---------------+--------------------+



## Total Revenue Analysis



In [21]:
from pyspark.sql.functions import sum

df.select(sum("revenue")).show()


+--------------------+
|        sum(revenue)|
+--------------------+
|2.1745103669900055E9|
+--------------------+



Conclusion

This analysis was performed using PySpark on a dataset containing 30,600 records to demonstrate scalable big data processing.

Key findings:

Total revenue generated: ₹2.17 Billion.

Premium Lifestyle category contributed the highest revenue.

Delhi NCR emerged as the top-performing state.

Revenue distribution between male and female customers is nearly equal.

Festival events significantly boost revenue during promotional periods.

Although the dataset size is moderate, the implementation using PySpark ensures scalability for handling millions of records efficiently in real-world scenarios.